include/picongpu/fields/laserProfiles/PulseFromSpectrum.hpp

```diff
@@ -0,0 +1,350 @@
+                        DataSpace<simDim> const& offsetToTotalDomain,
+                        float3_X const& elong,
+                        float_X const phase,
+                        uint32_t const currentStep)
+                        : m_elong(elong)
+                        , m_phase(phase)
+                        , m_currentStep(currentStep)
+                        , m_dataBoxE(dataBoxE)
+                        , m_offsetToTotalDomain(offsetToTotalDomain)
+                        , m_superCellToLocalOriginCellOffset(superCellToLocalOriginCellOffset)
+                    {
+                    }
+
+                    /** device side manipulation for init plane (transversal)
+                     *
+                     * @tparam T_Args type of the arguments passed to the user manipulator functor
+                     *
+                     * @param cellIndexInSuperCell ND cell index in current supercell
+                     */
+                    template<typename T_Acc>
+                    HDINLINE void operator()(T_Acc const&, DataSpace<simDim> const& cellIndexInSuperCell)
+                    {
+                        // coordinate system to global simulation as origin
+                        DataSpace<simDim> const localCell(cellIndexInSuperCell + m_superCellToLocalOriginCellOffset);
+
+                        // transform coordinate system to center of x-z plane of initialization
+                        constexpr uint8_t planeNormalDir = 1u;
+                        DataSpace<simDim> offsetToCenterOfPlane(m_offsetToTotalDomain);
+                        offsetToCenterOfPlane[planeNormalDir] = 0; // do not shift origin of plane normal
+                        floatD_X const pos
+                            = precisionCast<float_X>(localCell + offsetToCenterOfPlane) * cellSize.shrink<simDim>();
+                        // @todo add half-cells via traits::FieldPosition< Solver::NumicalCellType, FieldE >()
+
+                        // transversal position only
+                        floatD_X planeNoNormal = floatD_X::create(1.0_X);
+                        planeNoNormal[planeNormalDir] = 0.0_X;
+                        float_X const r2 = pmacc::math::abs2(pos * planeNoNormal);
+
+                        // position of the init plane relative to the focus position
+                        float_X const y_to_focus = pos.y() - Unitless::FOCUS_POS;
+
+                        // polarisation (only linear polarisation!)
+                        if(Unitless::Polarisation == Unitless::LINEAR_X)
+                        {
+                            m_elong.x() = E_t_dft(m_currentStep, r2, y_to_focus);
+                        }
+                        else if(Unitless::Polarisation == Unitless::LINEAR_Z)
+                        {
+                            m_elong.z() = E_t_dft(m_currentStep, r2, y_to_focus);
+                        }
+
+                        if(Unitless::initPlaneY != 0) // compile time if
```

Comment:
@sbastrakov I know this correction for cases where the initPlane is located at y != 0 used to be part of the standard gaussian as well. But it seems, this is now handled in a different way. Could you provide a description what to do now?